unit/src/HitTests.C

```diff
@@ -770,3 +773,202 @@ TEST(HitTests, vector_unsigned_int)
+  include_param_from_basefile << R"INPUT(
+  param01b = value01b
+)INPUT";
+  include_param_from_basefile.close();
+
+  // write block input to file on disk that is included from base input file
+  std::ofstream include_block_from_basefile("include_block_from_basefile.i");
+  include_block_from_basefile << R"INPUT(
+[Block02]
+  param02a = value02a
+  !include include_param_from_included.i
+  param02c = value02c
+[]
+)INPUT";
+  include_block_from_basefile.close();
+
+  // write param input to file on disk that is included from an include file
+  std::ofstream include_param_from_included("include_param_from_included.i");
+  include_param_from_included << R"INPUT(
+  param02b = value02b
+)INPUT";
+  include_param_from_included.close();
+
+  // parse the base input string to consume all contents from included files
+  auto root = hit::parse("BASE-STRING", basefile);
+
+  // delete the three files that were put on disk to be parsed from includes
+  std::remove("include_block_from_basefile.i");
+  std::remove("include_param_from_basefile.i");
+  std::remove("include_param_from_included.i");
+
+  // expected content from all include files when the parse tree is rendered
+  std::string render_expect = R"INPUT(
+[Block01]
+  param01a = value01a
+  param01b = value01b
+  param01c = value01c
+[]
+[Block02]
+  param02a = value02a
+  param02b = value02b
+  param02c = value02c
+[]
+[Block03]
+  param03a = value03a
+  param03b = value03b
+  param03c = value03c
+[]
+)INPUT";
+
+  // check that all file content is included when the parse tree is rendered
+  EXPECT_EQ(render_expect, "\n" + root->render() + "\n");
+
+  // expected content from all include files when the parse paths are listed
+  std::string tree_expect = R"INPUT(
+/                            - fname: BASE-STRING                     line:  2 column:  1
+/Block01                     - fname: BASE-STRING                     line:  2 column:  1
+/Block01/param01a (value01a) - fname: BASE-STRING                     line:  3 column:  3
+/Block01/param01b (value01b) - fname: ./include_param_from_basefile.i line:  2 column:  3
+/Block01/param01c (value01c) - fname: BASE-STRING                     line:  5 column:  3
+/Block02                     - fname: ./include_block_from_basefile.i line:  2 column:  1
+/Block02/param02a (value02a) - fname: ./include_block_from_basefile.i line:  3 column:  3
+/Block02/param02b (value02b) - fname: ./include_param_from_included.i line:  2 column:  3
+/Block02/param02c (value02c) - fname: ./include_block_from_basefile.i line:  5 column:  3
+/Block03                     - fname: BASE-STRING                     line:  8 column:  1
+/Block03/param03a (value03a) - fname: BASE-STRING                     line:  9 column:  3
+/Block03/param03b (value03b) - fname: BASE-STRING                     line: 10 column:  3
+/Block03/param03c (value03c) - fname: BASE-STRING                     line: 11 column:  3
```

Comment:
Is there any plan to be able to output this kind of information? At least where content actually came from. This would probably be pretty useful in debugging when we have a few levels of this stuff?

Reply:
@loganharbour -

I hadn't thought about a convenience method to dump this type of data.

I can add a `Node::info()` method to help with capturing this information

That's a good idea and could definitely be helpful for debugging.